sprint_7-solution-in-file-manager

7Jenko opened the pull request:
Добавить класс FileBackedTaskManager.
Добавить класс ManagerSaveException.

7Jenko committed:
sprint_7-solution-in-file-manager
Добавить класс FileBackedTaskManager.
Добавить класс ManagerSaveException.

7Jenko committed:
sprint_7-solution-in-file-manager
Исправить WhitespaceAround: '{' is not preceded with whitespace.

7Jenko committed:
sprint_7-solution-in-file-manager
Исправить WhitespaceAround: '{' is not preceded with whitespace.

slavrikk reviewed (commented):
Судя по ошибкам в билде, некоторые классы ты не закоммитил:
![image](https://github.com/user-attachments/assets/5c38ac05-4539-4c32-9181-044e6a270e1d)
Можно смотреть за результатом выполнения сборки, чтобы не допускать таких ситуаций)

7Jenko committed:
sprint_7-solution-in-file-manager
Исправить Build application.

7Jenko commented:
@slavrikk Извини, поторопился, не обратил внимания

slavrikk commented on src/com/yandex/app/service/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,166 @@
+        this.file = file;
+    }
+
+    public void save() {
```
Лучше private, так как это внутренняя реализация класса

slavrikk commented on src/com/yandex/app/service/InMemoryTaskManager.java:
```diff
@@ -25,23 +41,23 @@ public int generatorId() {
 
     @Override
-    public int addNewTask(Task task) {
+    public int addNewTask(Task task) throws IOException {
```
Во всех методах `throws IOException` лишнее

slavrikk commented on src/com/yandex/app/service/TaskManager.java:
```diff
@@ -4,16 +4,24 @@
 import java.util.List;
 
 public interface TaskManager {
+    ArrayList<Task> getAllTasks();
```
Лучше указать ссылку на интерфейс:
```
List<Task> getAllTasks();
List<Epic> getAllEpics();
List<Subtask> getAllSubtasks();
```

slavrikk commented on src/com/yandex/app/service/TaskManager.java:
```diff
@@ -4,16 +4,24 @@
     int generatorId();
 
-    int addNewTask(Task task);
+    int addNewTask(Task task) throws IOException;
```
Во всех методах throws IOException лишнее

slavrikk replied:
Замечание не исправлено для всех методов

7Jenko committed:
sprint_7-solution-in-file-manager
FileBackedTaskManager - исправить модификатор доступа метода save.
InMemoryTaskManager, TaskManager - убрать throws IOException.
TaskManager - указать ссылку на интерфейс.

7Jenko committed:
sprint_7-solution-in-file-manager
FileBackedTaskManager - исправить модификатор доступа метода save.
InMemoryTaskManager, TaskManager - убрать throws IOException.
TaskManager - указать ссылку на интерфейс.

slavrikk commented on src/com/yandex/app/service/InMemoryTaskManager.java:
```diff
@@ -11,12 +11,27 @@
-    private final Map<Integer, Task> tasks = new HashMap<>();
-    private final Map<Integer, Subtask> subtasks = new HashMap<>();
-    private final Map<Integer, Epic> epics = new HashMap<>();
+    protected static final Map<Integer, Task> tasks = new HashMap<>();
```
Не заметил в первый раз, лучше убрать static, чтобы у каждого объекта класса были свои коллекции тасок

7Jenko committed:
sprint_7-solution-in-file-manager
TaskManager - убрать throws IOException.
InMemoryTaskManager - убрать static.

7Jenko committed:
sprint_7-solution-in-file-manager
FileBackedTaskManagerTest - исправить failed test savingAndLoadingMultipleTasks.
